pkg/controller/installation/core_controller.go

```diff
@@ -640,6 +645,13 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		}
 	}
 
+	authentication, err := utils.GetAuthentication(ctx, r.client)
```

Comment:
I would have expected to see some kind of if authentication CRD exists here (perhaps like the AWSCloud one above).  I guess GetAuthentication() returns the default one?

Reply:
GetAuthentication() returns a struct for the authentication method found. It doesn't return the CRD, it parses the authentication CRD, determines the type (OIDC for right now), figures out any missing bits, and returns a struct with everything needed to setup the auth method. For OIDC this includes looking up the oidc secret to retrieve the client id and secret as well as parsing out the values we need from the well known config.